deploy_nixos/nixos-deploy.sh

```diff
@@ -20,6 +20,24 @@ sshOpts=(
   -o "GlobalKnownHostsFile=/dev/null"
 )
 
+###  Argument parsing ###
+
+drvPath="$1"
+targetHost="$2"
+sshPrivateKeyFile="$3"
+action="$4"
+shift
+shift
+shift
+shift
+# remove the last argument
+set -- "${@:1:$(($# - 1))}"
+buildArgs+=("$@")
+
+if [ -n "${sshPrivateKeyFile}" ]; then
+    sshOpts+=( -o "IdentityFile=${sshPrivateKeyFile}" )
+fi
```

Comment:
conceptually argument parsing is part of the main, not sure why it was moved up there

Reply:
The `sshOpts` needs the `sshPrivateKeyFile` argument now, so argparsing needs to come before the functions that use `sshOpts`.